tools/apprunner-gradle-plugin/src/main/java/com/dremio/nessie/quarkus/gradle/QuarkusApp.java

```diff
@@ -123,15 +123,15 @@ public static AppModel convert(Configuration configuration, Configuration deploy
     deploy.getResolvedConfiguration().getResolvedArtifacts().stream()
       .map(QuarkusApp::toDependency).forEach(deployDeps::add);
 
-
-    // find the path of the base app artifact
-    Optional<String> path = configuration.getFiles().stream().map(File::getAbsolutePath)
-      .filter(x->x.contains(appArtifact.getArtifactId()))
-      .filter(x->x.contains(appArtifact.getGroupId().replace(".", File.separator)))
-      .filter(x->x.contains(appArtifact.getVersion()))
-      .findFirst();
-    appArtifact.setPath(Paths.get(path.orElseThrow(() ->
-      new UnsupportedOperationException(String.format("Unknown path for app artifact %s", appArtifact)))));
+    // find the path of the base app artifact. We want the direct dependency only so we do a difference between all
+    // files and the the files we know from indirect dependencies. The leftovers are the files we want.
+    Set<File> files = configuration.getFiles();
```

Comment:
You can probably keep your streaming approach:

```
  final Set<File> userDepFiles = userDeps.stream().map(x -> x.getArtifact().getPaths().getSinglePath().toFile()).collect(Collectors.toSet());
  final Path path;
  try {
    path = configuration.getFiles()
      .filter(Predicates.not(userDepFiles::contains))
      .map(Paths::get)
      .collect(MoreCollectors.onlyElement());
  } catch (IllegalArgumentException e) {
    throw new UnsupportedOperationException(String.format("Unknown path for app artifact %s", appArtifact));
  }
```

Reply:
good idea, fixed